js/template.js

```diff
@@ -0,0 +1,65 @@
+const OBJECTS_TYPES = {
+  palace: 'Дворец',
+  flat: 'Квартира',
+  house: 'Дом',
+  bungalow: 'Бунгало',
+  hotel: 'Отель',
+};
+const cardTemplate = document
+  .querySelector('#card')
+  .content.querySelector('.popup');
+const mapCanvas = document.querySelector('.map__canvas');
+const fragment = document.createDocumentFragment();
+
+const fillTemplate = function ({ author, offer }) {
+  const newCard = cardTemplate.cloneNode(true);
+
+  const avatar = newCard.querySelector('.popup__avatar');
+  avatar.src = author.avatar;
```

Comment:
в принципе, для начала, хорошо.
В качестве задачки со звездочкой -подумай о том, что принято разделять куски кода

изготовить функцию - 

```js
const setAvatar = (author, img)=>img.src = author.avatar;
```
а тут
```suggestion
  setAvatar (author,  newCard.querySelector('.popup__avatar'));
```

что таким образом достигается? - экономия на названии переменной - трудно их придумывать
- и возможность добавлять сложности в "аватар" - проверка, что найдено, что это картинка, а не кнопка, например.